x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview_logic.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+const POLLING_DURATION = 5000;
+
+interface EngineOverviewApiData {
+  apiLogsUnavailable: boolean;
+  documentCount: number;
+  startDate: string;
+  endDate: string;
+  operationsPerDay: number[];
+  queriesPerDay: number[];
+  totalClicks: number;
+  totalQueries: number;
+}
+interface EngineOverviewValues extends EngineOverviewApiData {
+  dataLoading: boolean;
+  timeoutId: number | null;
+}
+
+interface EngineOverviewActions {
+  setPolledData(engineMetrics: EngineOverviewApiData): EngineOverviewApiData;
+  setTimeoutId(timeoutId: number): { timeoutId: number };
+  pollForOverviewMetrics(): void;
+  onPollingSuccess(engineMetrics: EngineOverviewApiData): EngineOverviewApiData;
+}
+
+export const EngineOverviewLogic = kea<MakeLogicType<EngineOverviewValues, EngineOverviewActions>>({
+  path: ['enterprise_search', 'app_search', 'engine_overview_logic'],
+  actions: () => ({
+    setPolledData: (engineMetrics) => engineMetrics,
```

Comment:
Non-blocking feedback. But it usually only takes a moment to do, so when I convert these logic files I often change actions to return objects based on Kea's recommendation:

> One more thing. It's strongly recommended to always return an object as a payload from your actions:

```suggestion
    setPolledData: (engineMetrics) => { engineMetrics },
```

Reply:
Oh this is super bizarre - I thought I left a comment on this but it appears to have disappeared :| So unfortunately based on the way the logic file is written we actually can't do this here without significantly rewriting / adding more destructuring for every single top-level value in the file. I haven't seen any other logic file do this, but in this file, we pull almost every top-level key out of the response and set it as its own value ([see the test](https://github.com/elastic/kibana/pull/83353/files#diff-bd22e69a4cee576e21c6bce2e477b8fdfda65ced191948599bf0e118c61a49aeR67).

Which means if we passed engineMetrics as an obj key, we now have to destructure that and do something like:

```js
         setPolledData: (_, { engineMetrics: { apiLogsUnavailable } }) => apiLogsUnavailable,
```

Which isn't the worst, but I went back and forth on this and ended up just deciding to leave it as it was from the current code. I definitely this this page and logic file could stand to be looked at first from a refactor POV in the future however, I believe it was one of the earliest ones created.

Reply:
Makes sense!